python/cudf/cudf/core/_base_index.py

```diff
@@ -608,19 +608,43 @@ def intersection(self, other, sort=False):
                     (1, 'Blue')],
                 )
         """
+        if not can_convert_to_column(other):
+            raise TypeError("Input must be Index or array-like")
+
         if not isinstance(other, BaseIndex):
-            other = cudf.Index(other, name=self.name)
+            other = cudf.Index(
+                other,
+                name=getattr(other, "name", self.name),
+            )
 
         if sort not in {None, False}:
             raise ValueError(
                 f"The 'sort' keyword only takes the values of "
                 f"None or False; {sort} was passed."
             )
 
-        if self.equals(other):
+        if self.equals(other) or not len(self):
```

Comment:
Switch these conditions to evaluate len(self) first. It should be cheaper and allow it to skip the equality check if empty.

Reply:
Done 👍